File: lib/models/user_profile.dart
class UserProfile {
  String ageRange;
  String profession;
  String industry;
  String degree;
  List<String> lifestyleTypes;
  String livingSituation;
  String energyPattern;
  int dailyFreeTime; // in minutes
  String stressBaseline;
  List<String> stressSources;
  String workloadIntensity;
  String motivationDriver;
  String failureResponse;
  String structurePreference;
  List<String> topValues;
  List<String> identityStatements;
  List<String> constraints;
  List<String> badHabits;
  String? currentLifePhase;

  UserProfile({
    this.ageRange = '',
    this.profession = '',
    this.industry = '',
    this.degree = '',
    this.lifestyleTypes = const [],
    this.livingSituation = '',
    this.energyPattern = 'morning',
    this.dailyFreeTime = 60,
    this.stressBaseline = 'medium',
    this.stressSources = const [],
    this.workloadIntensity = 'medium',
    this.motivationDriver = 'achievement',
    this.failureResponse = 'resilient',
    this.structurePreference = 'balanced',
    this.topValues = const [],
    this.identityStatements = const [],
    this.constraints = const [],
    this.badHabits = const [],
    this.currentLifePhase,
  });

  Map<String, dynamic> toJson() {
    return {
      'ageRange': ageRange,
      'profession': profession,
      'industry': industry,
      'degree': degree,
      'lifestyleTypes': lifestyleTypes,
      'livingSituation': livingSituation,
      'energyPattern': energyPattern,
      'dailyFreeTime': dailyFreeTime,
      'stressBaseline': stressBaseline,
      'stressSources': stressSources,
      'workloadIntensity': workloadIntensity,
      'motivationDriver': motivationDriver,
      'failureResponse': failureResponse,
      'structurePreference': structurePreference,
      'topValues': topValues,
      'identityStatements': identityStatements,
      'constraints': constraints,
      'badHabits': badHabits,
      'currentLifePhase': currentLifePhase,
    };
  }

  factory UserProfile.fromJson(Map<String, dynamic> json) {
    return UserProfile(
      ageRange: json['ageRange'] ?? '',
      profession: json['profession'] ?? '',
      industry: json['industry'] ?? '',
      degree: json['degree'] ?? '',
      lifestyleTypes: List<String>.from(json['lifestyleTypes'] ?? []),
      livingSituation: json['livingSituation'] ?? '',
      energyPattern: json['energyPattern'] ?? 'morning',
      dailyFreeTime: json['dailyFreeTime'] ?? 60,
      stressBaseline: json['stressBaseline'] ?? 'medium',
      stressSources: List<String>.from(json['stressSources'] ?? []),
      workloadIntensity: json['workloadIntensity'] ?? 'medium',
      motivationDriver: json['motivationDriver'] ?? 'achievement',
      failureResponse: json['failureResponse'] ?? 'resilient',
      structurePreference: json['structurePreference'] ?? 'balanced',
      topValues: List<String>.from(json['topValues'] ?? []),
      identityStatements: List<String>.from(json['identityStatements'] ?? []),
      constraints: List<String>.from(json['constraints'] ?? []),
      badHabits: List<String>.from(json['badHabits'] ?? []),
      currentLifePhase: json['currentLifePhase'],
    );
  }
}

File: backend/src/models/UserProfile.ts
import mongoose, { Schema } from "mongoose";
import {
  IUserProfile,
  EnergyPattern,
  StressLevel,
  WorkloadIntensity,
  MotivationDriver,
  FailureResponse,
  StructurePreference,
} from "../types";

const userProfileSchema = new Schema<IUserProfile>(
  {
    userId: {
      type: Schema.Types.ObjectId,
      ref: "User",
      required: true,
      unique: true,
      index: true,
    },
    ageRange: {
      type: String,
      default: "",
    },
    profession: {
      type: String,
      default: "",
      trim: true,
    },
    industry: {
      type: String,
      default: "",
      trim: true,
    },
    degree: {
      type: String,
      default: "",
      trim: true,
    },
    lifestyleTypes: {
      type: [String],
      default: [],
    },
    livingSituation: {
      type: String,
      default: "",
    },
    energyPattern: {
      type: String,
      enum: Object.values(EnergyPattern),
      default: EnergyPattern.Morning,
    },
    dailyFreeTime: {
      type: Number,
      default: 60,
      min: 0,
      max: 1440,
    },
    stressBaseline: {
      type: String,
      enum: Object.values(StressLevel),
      default: StressLevel.Medium,
    },
    stressSources: {
      type: [String],
      default: [],
    },
    workloadIntensity: {
      type: String,
      enum: Object.values(WorkloadIntensity),
      default: WorkloadIntensity.Medium,
    },
    motivationDriver: {
      type: String,
      enum: Object.values(MotivationDriver),
      default: MotivationDriver.Achievement,
    },
    failureResponse: {
      type: String,
      enum: Object.values(FailureResponse),
      default: FailureResponse.Resilient,
    },
    structurePreference: {
      type: String,
      enum: Object.values(StructurePreference),
      default: StructurePreference.Balanced,
    },
    topValues: {
      type: [String],
      default: [],
      validate: {
        validator: (values: string[]) => values.length <= 3,
        message: "Cannot have more than 3 top values",
      },
    },
    identityStatements: {
      type: [String],
      default: [],
    },
    constraints: {
      type: [String],
      default: [],
    },
    badHabits: {
      type: [String],
      default: [],
    },
    currentLifePhase: {
      type: String,
      default: null,
    },
  },
  {
    timestamps: true,
    toJSON: {
      transform(_doc, ret: Record<string, unknown>) {
        ret.id = ret._id;
        delete ret._id;
        delete ret.__v;
        return ret;
      },
    },
  },
);

const UserProfile = mongoose.model<IUserProfile>(
  "UserProfile",
  userProfileSchema,
);

export default UserProfile;
